orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,298 @@
-from nltk.stem.porter import PorterStemmer
+from nltk.corpus import stopwords
 from nltk.stem import WordNetLemmatizer
-from sklearn.feature_extraction.text import CountVectorizer
+from nltk.stem.porter import PorterStemmer
+from nltk.stem.snowball import SnowballStemmer
+from nltk.tokenize import word_tokenize
+
+from orangecontrib.text.corpus import Corpus
 
 
 class Preprocessor:
+    """A pre-processing class for the Orange text mining extension.
+
+    The class is capable of tokenizing, lowercasing, stemming and lemmatizing
+    the input. Returns a list of preprocessed tokens, sorted by order of
+    appearance in text.
     """
-        Holds pre-processing flags and other information, about stop word
-        removal, lowercasing, text morphing etc.(the options are set via
-        the Preprocess widget).
-    """
-    def __init__(self, incl_punct=False, lowercase=True, stop_words='english', trans=None, min_df=1):
+    def __init__(self, lowercase=True, stop_words=None,
+                 min_df=None, max_df=None, transformation=None,
+                 use_twitter_tokenizer=False):
         """
-        :param incl_punct: Determines whether the tokenizer should include punctuation in the tokens.
-        :type incl_punct: boolean
-        :param lowercase: If set, transform the tokens to lower case, before returning them.
-        :type lowercase: boolean
-        :param stop_words: Determines whether stop words should("english"), or should not(None) be removed.
-            If this is list, it should contain stopwords.
-        :type stop_words: 'english' or list or None
-        :param trans: An optional pre-processor object to perform the morphological
-            transformation on the tokens before returning them.
-        :type trans: :class: `orangecontrib.text.preprocess.Lemmatizer`
-            or :class: `orangecontrib.text.preprocess.Stemmer`
-        :return: :class: `orangecontrib.text.preprocess.Preprocessor`
+        Args:
+            lowercase (Optional[bool]): If set, transform the tokens to lower
+                case, before returning them. Defaults to False.
+            stop_words (Optional[str, list]) : Determines whether and what
+                stop words should be removed. Can remove default nltk stop
+                words for the English language or stop words provided in a
+                custom list. Defaults to None.
+            min_df (Optional[int, float]): Tokens that appear in exactly or
```

Comment:
I think this would be more intuitive if we cut of just the ones with strictly lower document frequencies. I suggest: `Tokens that appear in strictly less then min_df`. Same also for `max_df`.